### PR TITLE
tikv: forbid to try to get a connection forever

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve? 

`SendRequest` could be block forever if all connectons are broken.

### What is changed and how it works?

Forbid to try to get a connection forever so that `SendRequest` can return immediately and try an another target peer.

### Check List 

Tests 

 - unit tests

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: store/tikv/client_test.go [hunk 1]
```diff
@@ -22,6 +22,9 @@ import (
 	"github.com/pingcap/errors"
 	"github.com/pingcap/kvproto/pkg/tikvpb"
 	"github.com/pingcap/tidb/config"
+	"github.com/pingcap/tidb/store/tikv/tikvrpc"
+
+	"fmt"
```
Comment (lysu) [inline store/tikv/client_test.go]: need move import to first group

**Comment by lysu** [comment]
@hicqu  maybe better edit the description to follow TIDB PR template, our QA's tools will rely on PR template

**Comment by tiancaiamao** [comment]
LGTM

**Comment by jackysp** [comment]
Please fix CI.

**Pull request merged**